teacher.rb

```diff
@@ -1,10 +1,25 @@
-module Teacher
-  def initialize(specialization, age, parent_permission: true, name: 'Unknown')
-    super(name, age, parent_permission)
+require_relative './person'
+class Teacher < Person
+  @@teachers = []
+  def initialize(specialization, age, name, profession)
+    super(age, name, profession)
     @specialization = specialization
+    @@teachers.push(
+      {
+        'name' => name,
+        'specialization' => specialization,
+        'age' => age
+      }
+    )
```

Comment:
- [ ] @teachers array doesn't belong to a teacher object in OOP concept. So, kindly move it to appropriate based comments given in the Book class.